doc/topics/releases/2015.8.4.rst

```diff
@@ -0,0 +1,15 @@
+===========================
+Salt 2015.8.4 Release Notes
+===========================
+
+Core Changes
+============
+
+- Added ``state.validate`` feature.  This checks SLS data against the salt
+  state API and reports errors if the argument types do not match the types
```

Comment:
This iteration cannot check types for the reason Mike mentioned below. In order to do type checking we need to implement the structured docstring parsing that we whiteboarded last month.

For this first release we can only check argument names plus a few common state constructions (like `requires` -> `require`).

If the state function accepts `**kwargs` then it reduces our ability to check even argument names since the Voluptuous schema will necessarily need to accept all values. But this is also something we can improve in a later iteration. (I have a few notes on this we can put into a new issue.)